android/src/ti/map/ViewProxy.java

```diff
@@ -780,7 +789,50 @@ public void handleAddPolyline(Object polyline) {
 		} else {
 			addPreloadPolyline(p);
 		}
+	}
+
+	@Kroll.method
+	public void addPolylines(Object polylinesObject) {
+		if (!(polylinesObject instanceof Object[])) {
+			Log.e(TAG, "Invalid argument to addPolylines",  Log.DEBUG_MODE);
+			return;
+		}
+		Object[] lines = (Object[])polylinesObject;
 
+		//Update the JS object
+		Object polylines = getProperty(MapModule.PROPERTY_POLYLINES);
+		if (polylines instanceof Object[]) {
+			ArrayList<Object> polylinesList = new ArrayList<Object>(
+					Arrays.asList((Object[]) polylines));
+			for (int i = 0; i < lines.length; i++) {
+				Object polylineObject = lines[i];
+				if (polylineObject instanceof AnnotationProxy) {
+					polylinesList.add(polylineObject);
+				}
+			}
+			setProperty(MapModule.PROPERTY_POLYLINES, polylinesList.toArray());
+		} else {
+			setProperty(MapModule.PROPERTY_POLYLINES, lines);
+		}
+
+		TiUIView view = peekView();
+		if (view instanceof TiUIMapView) {
+			if (TiApplication.isUIThread()) {
+				handleAddAnnotations(lines);
```

Comment:
Shouldn't this be `handleAddPolylines`?

Reply:
You are absolutely right.... copy & paste error!